connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/ActivatedExecutableManager.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.executable;
+
+import io.camunda.connector.api.error.ConnectorRetryException;
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorFactory;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorReportingContext;
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ActivatedExecutableManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ActivatedExecutableManager.class);
+  private final InboundConnectorFactory inboundConnectorFactory;
+  private final InboundConnectorReportingContext context;
+  private final ScheduledExecutorService reactivationExecutor;
+  boolean isCancelled;
+  private InboundConnectorExecutable<InboundConnectorContext> executable;
+
+  public ActivatedExecutableManager(
```

Comment:
No, you are right, something in fundamentally flawed here. The issue is that we have to deal with multiple things:

- We have to keep track of the status of an executable (cancelled or activated) in the `InboundExecutableRegistry` because it publishes the status of every connector, without this, we would not know the status
- cancel can only be called from the context, and to cancel properly; we need access to the executable
- The only place we could cancel was somewhere we have both access to the context AND the executable,

Those solution tries to answer this: 
- The creation of ActivatedExecutableManager that wraps the executable AND the context, being itself wrapped into registeredExecutable to be accessible from `InboundExecutableRegistry`
- This class keeps track of the status of the connector (isCancelled) and therefore is accessible from `InboundExecutableRegistry`
- We pass the Consumer to the context, so it can call it when context is called

The main complexity is that we have `InboundExecutableRegistryImpl` which contains a map, which contains an RegiteredExecutable containing the context and the executable, that both of them are needed to cancel, and that cancellation needs to have side effect known by the `InboundExecutableRegistryImpl`